Fails to install with stable rustc (1.71.1) and cargo (0.72.2)

Our CI started failing with latest stable versions of the toolchain installed by rustup.

Error:
```
   ...
   Compiling gix-discover v0.18.1
   Compiling gix-config v0.22.0
   Compiling gix-transport v0.31.0
   Compiling crates-io v0.37.0
   Compiling gix-protocol v0.32.0
   Compiling gix v0.44.1
   Compiling git2 v0.17.2
   Compiling git2-curl v0.18.0
error[E0599]: `Vec<u8>` is not an iterator
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-0.72.2/src/cargo/core/compiler/future_incompat.rs:275:18
    |
274 | /             strip_ansi_escapes::strip(&to_display)
275 | |                 .map(|v| String::from_utf8(v).expect("utf8"))
    | |                 -^^^ `Vec<u8>` is not an iterator; try calling `.into_iter()` or `.iter()`
    | |_________________|
    | 
   --> /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/alloc/src/vec/mod.rs:396:1
    |
    = note: doesn't satisfy `Vec<u8>: Iterator`
    |
    = note: the following trait bounds were not satisfied:
            `Vec<u8>: Iterator`
            which is required by `&mut Vec<u8>: Iterator`
            `[u8]: Iterator`
            which is required by `&mut [u8]: Iterator`
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
36  + use winnow::parser::Parser;
    |

error[E0599]: `Vec<u8>` is not an iterator
    --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-0.72.2/src/cargo/core/compiler/mod.rs:1644:26
     |
1643 | /                     strip_ansi_escapes::strip(&msg.rendered)
1644 | |                         .map(|v| String::from_utf8(v).expect("utf8"))
     | |                         -^^^ `Vec<u8>` is not an iterator; try calling `.into_iter()` or `.iter()`
     | |_________________________|
     | 
    --> /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/alloc/src/vec/mod.rs:396:1
     |
     = note: doesn't satisfy `Vec<u8>: Iterator`
     |
     = note: the following trait bounds were not satisfied:
             `Vec<u8>: Iterator`
             which is required by `&mut Vec<u8>: Iterator`
             `[u8]: Iterator`
             which is required by `&mut [u8]: Iterator`
     = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
     |
34   + use winnow::parser::Parser;
     |

error[E0599]: `Vec<u8>` is not an iterator
    --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-0.72.2/src/cargo/core/compiler/mod.rs:1678:22
     |
1677 |                   error.rendered = strip_ansi_escapes::strip(&error.rendered)
     |  __________________________________-
1678 | |                     .map(|v| String::from_utf8(v).expect("utf8"))
     | |                     -^^^ `Vec<u8>` is not an iterator; try calling `.into_iter()` or `.iter()`
     | |_____________________|
     | 
    --> /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/alloc/src/vec/mod.rs:396:1
     |
     = note: doesn't satisfy `Vec<u8>: Iterator`
     |
     = note: the following trait bounds were not satisfied:
             `Vec<u8>: Iterator`
             which is required by `&mut Vec<u8>: Iterator`
             `[u8]: Iterator`
             which is required by `&mut [u8]: Iterator`
     = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
     |
34   + use winnow::parser::Parser;
     |

For more information about this error, try `rustc --explain E0599`.
error: could not compile `cargo` (lib) due to 3 previous errors
error: failed to compile `cargo-c v0.9.22+cargo-0.72`, intermediate artifacts can be found at `/tmp/cargo-installSEPbgS`
```

I used this command to reproduce it as of 8 Aug 2023:
```sh
docker run --rm ubuntu:22.04 bash -c 'apt-get update && apt-get install -y curl build-essential libssl-dev pkg-config && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y && env PATH="/root/.cargo/bin:${PATH}" cargo install cargo-c'
```

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -23,7 +23,6 @@ jobs:
       - name: Run clippy
         uses: actions-rs-plus/clippy-check@v2.0.0
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
           args: --all -- -D warnings --verbose
 
   coverage:

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "cargo-c"
-version = "0.9.22+cargo-0.72"
+version = "0.9.23+cargo-0.72.2"
 authors = ["Luca Barbato <lu_zero@gentoo.org>"]
 description = "Helper program to build and install c-like libraries"
 license = "MIT"
@@ -28,7 +28,7 @@ name = "cargo-ctest"
 path = "src/bin/ctest.rs"
 
 [dependencies]
-cargo = "0.72.1"
+cargo = "0.72.2"
 cargo-util = "0.2"
 semver = "1.0.3"
 log = "0.4"
@@ -43,6 +43,8 @@ anyhow = "1.0"
 cc = "1.0"
 glob = "0.3"
 itertools = "0.11"
+# pin a cargo dependency that sadly broke semver
+strip-ansi-escapes = "=0.1.1"
 
 [features]
 default = []

